crypto/store/loader_file.c

```diff
@@ -0,0 +1,1027 @@
+                } else {
+                    /* more than one match => ambiguous, kill any result */
+                    STORE_INFO_free(result);
+                    STORE_INFO_free(tmp_result);
+                    if (handler->destroy_ctx != NULL)
+                        handler->destroy_ctx(&handler_ctx);
+                    handler_ctx = NULL;
+                    result = NULL;
+                }
+            }
+        }
+
+        if (*matchcount > 1)
+            STOREerr(STORE_F_FILE_LOAD_TRY_DECODE,
+                     STORE_R_AMBIGUOUS_CONTENT_TYPE);
+        if (*matchcount == 0)
+            STOREerr(STORE_F_FILE_LOAD_TRY_DECODE,
+                     STORE_R_UNSUPPORTED_CONTENT_TYPE);
+        else if (matching_handlers[0]->repeatable) {
+            if (ctx == NULL) {
+                STOREerr(STORE_F_FILE_LOAD_TRY_DECODE,
+                         STORE_R_UNSUPPORTED_CONTENT_TYPE);
+                STORE_INFO_free(result);
+                result = NULL;
+            } else {
+                ctx->_.file.last_handler = matching_handlers[0];
+                ctx->_.file.last_handler_ctx = handler_ctx;
+            }
+        }
+
+        OPENSSL_free(matching_handlers);
+    }
+
+ err:
+    if (new_pem_name != NULL)
+        OPENSSL_free(new_pem_name);
+    if (new_mem != NULL)
+        BUF_MEM_free(new_mem);
+
+    if (result != NULL
+        && (t = STORE_INFO_get_type(result)) == STORE_INFO_DECODED) {
+        pem_name = new_pem_name = store_info_get0_DECODED_pem_name(result);
+        new_mem = store_info_get0_DECODED_buffer(result);
+        data = (unsigned char *)new_mem->data;
+        len = new_mem->length;
+        OPENSSL_free(result);
+        result = NULL;
+        goto again;
+    }
+
+    if (result != NULL)
+        ERR_clear_error();
+
+    return result;
+}
+
+static STORE_INFO *file_load_try_repeat(STORE_LOADER_CTX *ctx,
+                                        const UI_METHOD *ui_method,
+                                        void *ui_data)
+{
+    STORE_INFO *result = NULL;
+
+    if (ctx->_.file.last_handler != NULL) {
+        result =
+            ctx->_.file.last_handler->try_decode(NULL, NULL, NULL, 0,
+                                                 &ctx->_.file.last_handler_ctx,
+                                                 ui_method, ui_data);
+
+        if (result == NULL) {
+            ctx->_.file.last_handler->destroy_ctx(&ctx->_.file.last_handler_ctx);
+            ctx->_.file.last_handler_ctx = NULL;
+            ctx->_.file.last_handler = NULL;
+        }
+    }
+    return result;
+}
+
+static int file_read_pem(BIO *bp, char **pem_name, char **pem_header,
+                         unsigned char **data, long *len,
+                         const UI_METHOD *ui_method,
+                         void *ui_data)
+{
+    int i = PEM_read_bio(bp, pem_name, pem_header, data, len);
+
+    if (i <= 0)
+        return 0;
+
+    if (strlen(*pem_header) > 10) {
```

Comment:
Oh great, you're asking me to explain *someone else's arbitrary number* that's been in the source since the SSLeay days ;-)

I *think*, however, that the explanation lies [here](https://github.com/openssl/openssl/blob/master/crypto/pem/pem_lib.c#L494-L496), with the header keys we recognise, and that are about 10 characters each, so anything shorter than that can't be a valid header for OpenSSL use and is therefore ignored.  I know, it's a pretty lame reason, but if we're going to venture a guess, that would be mine.

If it were me, I could just as well have `> 0`, although I'm unsure about the implications, so I think I'll go for the comment idea.